src/ui/preview/preview.js

```diff
@@ -463,11 +469,56 @@ cwc.ui.Preview.prototype.focus = function() {
 /**
  * Injects and executes the passed code in the preview content, if supported.
  * @param {!(string|Function)} code
+ * @param {Function=} callback
  */
-cwc.ui.Preview.prototype.executeScript = function(code) {
+cwc.ui.Preview.prototype.executeScript = function(code, callback) {
```

Comment:
The Preview should not be in charge of handling the callback.
It should only "send and forget" stuff, and other parts like the messenger needs to make sure that the callback is handled correctly.